model/labels/labels.go

```diff
@@ -311,6 +312,19 @@ func (ls Labels) WithoutEmpty() Labels {
 	return ls
 }
 
+// IsValid checks if the metric name or label names are valid.
+func (ls Labels) IsValid() bool {
+	for _, l := range ls {
+		if !model.LabelName(l.Name).IsValid() {
+			return false
+		}
```

Comment:
The label value can also be invalid

Reply:
You are right, but I didn't find this in the document.

Reply:
What do you mean?

Reply:
I changed the order, I mean this doc https://prometheus.io/docs/concepts/data_model/

Reply:
Oooh you added both on the same if.